cl.hpp missing - OpenCL version mismatch?

Hello everyone, 
I was trying to run the GEMM benchmark, cmake configuration succeeds but building fails with error CL/cl.hpp, no such file or directory. In my /usr/include/CL directory i don't have such file. May this be due to a mismatch in our OpenCL versions? My cl2.hpp file mentions version 2.0.7.
I have also trying setting USE_DEPRECATED_CPP_HEADER to false, in this case it fails with a sintax error in cl2.hpp, line 7841,  expected ; at the end of method declaration.
Thanks
Pietro

Hello,
thanks for your reply. After your changes I still cannot build. I get the following error.
`In file included from /usr/include/CL/cl.h:30:0,
                 from /usr/include/CL/opencl.h:42,
                 from /usr/include/CL/cl2.hpp:453,
                 from /home/fpga/pietro/hpccv2/HPCC_FPGA/shared/include/setup/fpga_setup.hpp:37,
                 from /home/fpga/pietro/hpccv2/HPCC_FPGA/shared/setup/fpga_setup.cpp:5:
/usr/include/CL/cl2.hpp:7841:30: error: expected ‘;’ at end of member declaration
         Event* event = NULL) CL_EXT_SUFFIX__VERSION_1_2_DEPRECATED const
                              ^
In file included from /home/fpga/pietro/hpccv2/HPCC_FPGA/shared/include/setup/fpga_setup.hpp:37:0,
                 from /home/fpga/pietro/hpccv2/HPCC_FPGA/shared/setup/fpga_setup.cpp:5:
/usr/include/CL/cl2.hpp:7842:5: error: expected unqualified-id before ‘{’ token
     {`

Looking at cl2.hpp, the line that raises the error appears after
`#if defined(CL_USE_DEPRECATED_OPENCL_1_2_APIS)`

Your project doesn't seem to set this flag, but by running `grep -r CL_USE_DEPRECATED_OPENCL_1_2_APIS`, I have found the following:
`build/GEMM/_deps/extern_hlslib-src/include/hlslib/xilinx/SDAccel.h:#define CL_USE_DEPRECATED_OPENCL_1_2_APIS`.

At this point I don't know if it something that you can fix on your side. I have manged to compile by manually adding the cl.hpp file in my include dir, so feel free to close the issue if you find it appropriate.
Thanks

--- a/GEMM/CHANGELOG
+++ b/GEMM/CHANGELOG
@@ -2,6 +2,11 @@
 
 This file contains all changes made to the source code for each release.
 
+## 1.3
+
+## Added:
+- Support for cl2.hpp header file by removing hard links to cl.hpp
+
 ## 1.2
 
 ## Added:

--- a/GEMM/CMakeLists.txt
+++ b/GEMM/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.13)
-project(GEMM VERSION 1.2)
+project(GEMM VERSION 1.3)
 
 set(KERNEL_NAME gemm CACHE STRING "Name of the OpenCL kernel")
 set(DEFAULT_MATRIX_SIZE 8 CACHE STRING "Default size of the used matrices")

--- a/GEMM/src/host/execution.h
+++ b/GEMM/src/host/execution.h
@@ -26,9 +26,6 @@ SOFTWARE.
 #include <memory>
 #include <vector>
 
-/* External library headers */
-#include "CL/cl.hpp"
-
 #include "parameters.h"
 #include "gemm_benchmark.hpp"
 

--- a/GEMM/src/host/execution_default.cpp
+++ b/GEMM/src/host/execution_default.cpp
@@ -30,7 +30,6 @@ SOFTWARE.
 #include <vector>
 
 /* External library headers */
-#include "CL/cl.hpp"
 #ifdef INTEL_FPGA
 #include "CL/cl_ext_intelfpga.h"
 #endif
@@ -215,7 +214,7 @@ calculate(hpcc_base::ExecutionSettings<gemm::GEMMProgramSettings> const& config,
 #endif
         auto t1 = std::chrono::high_resolution_clock::now();
         for (int i=0; i < config.programSettings->kernelReplications; i++) {
-            compute_queues[i].enqueueTask(gemmkernels[i]);
+            compute_queues[i].enqueueNDRangeKernel(gemmkernels[i], cl::NullRange, cl::NDRange(1));
         }
         for (int i=0; i < config.programSettings->kernelReplications; i++) {
             compute_queues[i].finish();

